crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_structure.rs

```diff
@@ -67,23 +57,22 @@ pub(crate) const fn row_and_column_from_index(index: usize) -> (usize, usize) {
 
 /// Returns the index of data where the (row, column) belongs.
 #[allow(dead_code)]
-pub(crate) const fn index_from_row_and_column(row: usize, column: usize) -> usize {
+pub(crate) fn index_from_row_and_column(row: usize, column: usize) -> Option<usize> {
     let width_of_row = full_width_of_row(row);
-
-    // Return no_index if the (row, column) is not in dynamic Dory structure.
-    if column >= width_of_row || (row == 1 && column == 0) {
-        return no_index();
-    }
-
-    width_of_row * (row - width_of_row / 2) + column
+    (column < width_of_row && (row, column) != (1, 0))
+        .then_some(width_of_row * (row - width_of_row / 2) + column)
 }
 
-/// Returns a matrix size that can hold the given number of data points being committed with respect to an offset.
+/// Returns a matrix size, (height, width), that can hold the given number of data points being committed with respect to an offset.
 #[allow(dead_code)]
 pub(crate) const fn matrix_size(data_len: usize, offset: usize) -> (usize, usize) {
+    if data_len == 0 {
+        return (0, 0);
+    }
```

Comment:
Is this consistent with the other methods?
This is certainly an edge case that I would document.

Reply:
I feel like this is mixing two concepts: the "trimmed" matrix (where all the top rows made up of just 0s caused by the offset are removed) and the "untrimmed" matrix.

Both may be useful, and I'm not sure which one you actually want.

What this is doing, when `data_len` is 0, it is always the "trimmed" size. Otherwise, it is the "untrimmed" size. This is confusing and could lead to upstream bugs or more complex code.

Reply:
`matrix_size` needs to make sure both `data_len` and `offset` aren't `0` to not panic at the line `row_and_column_from_index(offset + data_len - 1)`. My thought was if a caller asked what size of a matrix they need to hold no data, i.e. `data_len == 0`, it was a degenerate case and the result should be an empty matrix regardless of the offset. I see your point about mixing trimmed and untrimmed matrices.

I've updated the code to only return `(0, 0)` when `data_len == offset == 0`. So the method now returns an untrimmed matrix size. I've also added a note to the doc.